hal/ia32/memory.c

```diff
@@ -195,28 +195,64 @@ int hal_memoryGetNextEntry(addr_t start, addr_t end, mapent_t *entry)
 
 
 /* Flushes 8042 keyboard controller buffers */
-static void memory_empty8042(void)
+static int memory_empty8042(void)
 {
 	unsigned char status;
-
-	do {
+	unsigned short int timeout;
+	for (timeout = 0xffff; timeout > 0; --timeout) {
```

Comment:
`timeout` is never `<0`, because it's `unsigned`
```suggestion
	for (timeout = 0xffff; timeout != 0; --timeout) {
```

Reply:
It's ok as is, timeout > 0 means the same thing as timeout != 0 in thus case